src/linuxDistribution.ts

```diff
@@ -0,0 +1,67 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as fs from 'fs';
+import * as os from 'os';
+
+export class LinuxDistribution {
+    constructor(public name: string, public version: string) { }
+
+    /**
+     * There is no standard way on Linux to find the distribution name and version.
+     * Recently, systemd has pushed to standardize the os-release file. This has
+     * seen adoption in "recent" versions of all major distributions.
+     * https://www.freedesktop.org/software/systemd/man/os-release.html
+     */
+    public static GetDistroInformation(): Promise<LinuxDistribution> {
+        let linuxDistro: Promise<LinuxDistribution>;
+
+        // First check /etc/os-release and only fallback to /usr/lib/os-release
+        // as per the os-release documentation.
+        linuxDistro = LinuxDistribution.getDistroInformationFromFile('/etc/os-release')
+            .catch(() => {
+                return LinuxDistribution.getDistroInformationFromFile('/usr/lib/os-release');
+            }).catch(() => {
+                // couldn't get distro information
+                return Promise.resolve(new LinuxDistribution('unknown', 'unknown'));
+            });
+
+        return linuxDistro;
```

Comment:
Any reason for assigning this to a local `let` instead of returning immediately?